addon/es2015-class.js

```diff
@@ -0,0 +1,27 @@
+import { core, update, destroy } from './core';
+import { inject } from '@ember/service';
+
+export default function wrapEs2015Class(stateToComputed, dispatchToActions, WrappedComponent) {
+
+  WrappedComponent.prototype.redux = inject('redux');
+
+  return class Connect extends WrappedComponent {
+
+    constructor() {
+      super(...arguments);
+      core.call(this, stateToComputed, dispatchToActions);
+    }
+
+    didUpdateAttrs() {
+      super.didUpdateAttrs(...arguments);
```

Comment:
You probably will want to put this call to `didUpdateAttrs` in a guard, since the superclass may or may not have implemented `didUpdateAttrs`. We need to figure out what the solution is long term is Ember for this